apps/homescreen/test/unit/app_test.js

```diff
@@ -1,36 +1,564 @@
+        assert.isFalse(scrollToSy.called);
+
+        app.scrollable.scrollTop = 50;
+        app.snapScrollPosition(0);
+        assert.isTrue(scrollToSy.called);
+      });
+    });
+  });
+
+  suite('App#showActionDialog()', () => {
+    var clock;
+
+    setup(function() {
+      clock = sinon.useFakeTimers();
+    });
+
+    teardown(function() {
+      clock.restore();
+    });
+
+    test('should attach events on click', () => {
+      var dialog = app.settingsDialog;
+      dialog.open = () => {
+      };
+      app.showActionDialog(dialog, null, [() => {
+      }]);
+      assert.ok(dialog.querySelector('button').onclick);
+    });
+
+    test('should set the title of the dialog', () => {
+      var dialog = app.cancelDownload;
+      dialog.open = () => {
+      };
+      assert.isFalse(dialog.querySelector('.body')
+        .hasAttribute('data-l10n-args'));
+      app.showActionDialog(dialog, 'abc', []);
+      assert.isTrue(dialog.querySelector('.body')
+        .hasAttribute('data-l10n-args'));
+      clock.tick(100);
+    });
+
+    test('should show the dialog after a little while', done => {
+      var dialog = {
+        style: { display: 'none' },
+        getElementsByClassName: () => [],
+        querySelector: () => {
+        },
+        open: () => {
+          done();
+        }
+      };
+
+      app.showActionDialog(dialog, null, []);
+      clock.tick(100);
+    });
+  });
+
+  suite('App#handleEvent()', () => {
+    suite('contextmenu', () => {
+      test('should open a dialog', () => {
+        var showActionDialogStub = sinon.stub(app, 'showActionDialog');
+        app.handleEvent(new CustomEvent('contextmenu'));
+        assert.isTrue(showActionDialogStub.called);
+        //showActionDialogStub.restore();
+      });
+
+      test('should attach a configure web activity to a button click', () => {
+        app.settingsDialog = {
+          open: () => {
+          },
+          close: () => {
+          }
+        };
+        var showActionDialogStub = sinon.stub(app, 'showActionDialog',
+          (dialog, args, callbacks) => {
+            callbacks[0]();
+          });
+        app.handleEvent(new CustomEvent('contextmenu'));
+        assert.isTrue(showActionDialogStub.called);
+        assert.equal(MockMozActivity.calls.length, 1);
+        assert.equal(MockMozActivity.calls[0].name, 'configure');
+        //showActionDialogStub.restore();
+      });
+    });
+
+    suite('scroll', () => {
+      test('should show and hide the drop shadow accordingly', () => {
+        assert.isFalse(app.scrolled);
+        app.scrollable = {
+          scrollTop: 50
+        };
+        app.handleEvent(new CustomEvent('scroll'));
+        assert.isTrue(app.scrolled);
+        assert.equal(app.shadow.className, 'visible');
+
+        app.scrollable.scrollTop = 0;
+        app.handleEvent(new CustomEvent('scroll'));
+        assert.isFalse(app.scrolled);
+        assert.equal(app.shadow.className, '');
```

Comment:
It might be nicer to use assert.isTrue/False(classList.contains('scroll')) here? I don't imagine we'll be adding other classes to this, but just in case.